specs/cn/v1.0.0-rc.2.md

```diff
@@ -0,0 +1,915 @@
+```
+
+值必须是下述类型之一。  
+
+- [字符串](#字符串)
+- [整数](#整数)
+- [浮点数](#浮点数)
+- [布尔值](#布尔值)
+- [坐标日期时刻](#坐标日期时刻)
+- [各地日期时刻](#各地日期时刻)
+- [各地日期](#各地日期)
+- [各地时刻](#各地时刻)
+- [数组](#数组)
+- [内联表](#内联表)
+
+未指定值是非法的。  
+
+```toml
+key = # 非法
+```
+
+键值对后必须换行（或 EOF）。（例外见 [内联表](#内联表)）  
+
+```
+first = "Tom" last = "Preston-Werner" # 非法
+```
+
+键名
+----
+
+键名可以是裸露的，引号引起来的，或点分隔的。  
+
+**裸键**只能包含 ASCII 字母，ASCII 数字，下划线和短横线（`A-Za-z0-9_-`）。  
+注意裸键允许仅由纯 ASCII 数字构成，例如 `1234`，但是是被理解为字符串的。  
+
+```toml
+key = "value"
+bare_key = "value"
+bare-key = "value"
+1234 = "value"
+```
+
+**引号键**遵循与基本字符串或字面量字符串相同的规则并允许你使用更为广泛的键名。  
+除非明显必要，使用裸键方为最佳实践。  
+
+```toml
+"127.0.0.1" = "value"
+"character encoding" = "value"
+"ʎǝʞ" = "value"
+'key2' = "value"
+'quoted "value"' = "value"
+```
+
+裸键中不能为空，但空引号键是允许的（虽然不建议如此）。  
+
+```toml
+= "no key name"  # 非法
+"" = "blank"     # 合法但不鼓励
+'' = 'blank'     # 合法但不鼓励
+```
+
+**点分隔键**是一系列通过点相连的裸键或引号键。  
+这允许了你将相近属性放在一起：  
+
+```toml
+name = "Orange"
+physical.color = "orange"
+physical.shape = "round"
+site."google.com" = true
+```
+
+等价于 JSON 的如下结构：
+
+```json
+{
+  "name": "Orange",
+  "physical": {
+    "color": "orange",
+    "shape": "round"
+  },
+  "site": {
+    "google.com": true
+  }
+}
+```
+
+点分隔符周围的空白会被忽略，不过，最佳实践是不要使用任何不必要的空白。  
+
+多次定义同一个键是不行的。  
+
+```
+# 不要这样做
+name = "Tom"
+name = "Pradyun"
+```
+
+注意，裸键和引号键是等效的：
+
+```
+# 这将不起作用
+spelling = "favorite"
+"spelling" = "favourite"
+```
+
+只要一个键还没有被直接定义过，你就仍可以对它和它下属的键名赋值。  
+
+```
+# 这使“fruit”键作为表存在。
+fruit.apple.smooth = true
+
+# 所以接下来你可以像中这样对“fruit”表添加内容：
+fruit.orange = 2
+```
+
+```
+# 以下是非法的
+
+# 这将 fruit.apple 的值定义为一个整数。
+fruit.apple = 1
+
+# 但接下来这将 fruit.apple 像表一样对待了。
+# 整数不能变成表。
+fruit.apple.smooth = true
+```
+
+不鼓励跳跃式地定义点分隔键。  
+
+```toml
+# 合法但不鼓励
+
+apple.type = "水果"
+orange.type = "水果"
+
+apple.skin = "薄"
+orange.skin = "厚"
+
+apple.color = "红"
+orange.color = "橙"
+```
+
+```toml
+# 建议
+
+apple.type = "水果"
+apple.skin = "薄"
+apple.color = "红"
+
+orange.type = "水果"
+orange.skin = "厚"
+orange.color = "橙"
+```
+
+由于裸键只允许由 ASCII 整数组成，所以可能写出看起来像浮点数、但实际上是两部分的点分隔键。  
+除非你有充分的理由（基本不太会），否则不要这样做。  
+
+```toml
+3.14159 = "派"
+```
+
+上面的 TOML 对应以下 JSON。  
+
+```json
+{ "3": { "14159": "派" } }
+```
+
+字符串
+------
+
+共有四种方式来表示字符串：基本字符串、多行基本字符串、字面量和多行字面量。  
+所有字符串都只能包含有效的 UTF-8 字符。  
+
+**基本字符串**由引号（`"`）包裹。  
+任何 Unicode 字符都可以使用，除了那些必须转义的：引号，反斜杠，以及除制表符外的控制字符（U+0000 至 U+0008，U+000A 至 U+001F，U+007F）。  
+
+```toml
+str = "我是一个字符串。\"你可以把我引起来\"。姓名\t乔斯\u00E9\n位置\t旧金山。"
+```
+
+为了方便，一些流行的字符有其简便转义写法。
+
+```
+\b         - backspace       (U+0008)
+\t         - tab             (U+0009)
+\n         - linefeed        (U+000A)
+\f         - form feed       (U+000C)
+\r         - carriage return (U+000D)
+\"         - quote           (U+0022)
+\\         - backslash       (U+005C)
+\uXXXX     - unicode         (U+XXXX)
+\UXXXXXXXX - unicode         (U+XXXXXXXX)
+```
+
+任何 Unicode 字符都可以用 `\uXXXX` 或 `\UXXXXXXXX` 的形式来转义。  
+转义码必须是有效的 Unicode [标量值](http://unicode.org/glossary/#unicode_scalar_value)。  
```

Comment:
```suggestion
转义码必须是有效的 Unicode [标量值](https://unicode.org/glossary/#unicode_scalar_value)。  
```

Reply:
realized this is the upstream issue, submit over [there](https://github.com/toml-lang/toml/pull/786).